### ISSUE TITLE
Eda

### ISSUE THREAD

**Issue by Nithin7983** [Nithin7983/Dataset-and-preprocessing-#3]
# Titanic EDA Script (Cleaned and Improved)

import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt
import plotly.express as px
from sklearn.preprocessing import StandardScaler

# 1. Load dataset
df = pd.read_csv("titanic.csv")

# 2. Basic Info
print("First 5 rows:")
print(df.head())
print("\nDataset Info:")
df.info()
print("\nMissing Values:")
print(df.isnull().sum())

# 3. Handle missing values
df['Age'].fillna(df['Age'].median(), inplace=True)
df['Embarked'].fillna(df['Embarked'].mode()[0], inplace=True)
df.drop(columns=['Cabin'], inplace=True, errors='ignore')

# 4. Encode categorical variables
df['Sex'] = df['Sex'].map({'male': 0, 'female': 1})
df = pd.get_dummies(df, columns=['Embarked'], drop_first=True)

# 5. Feature scaling for Age and Fare
scaler = StandardScaler()
df[['Age', 'Fare']] = scaler.fit_transform(df[['Age', 'Fare']])

# 6. Remove duplicates
df.drop_duplicates(inplace=True)

# 7. Summary statistics
print("\nSummary Statistics:")
print(df.describe())

# 8. Histograms for numeric features
numeric_features = ['Age', 'Fare', 'SibSp', 'Parch']
df[numeric_features].hist(bins=20, figsize=(12, 8), color='skyblue', edgecolor='black')
plt.suptitle("Histograms of Numeric Features", y=1.02)
plt.tight_layout()
plt.show()

# 9. Boxplots for numeric features
plt.figure(figsize=(12, 6))
for i, feature in enumerate(numeric_features):
    plt.subplot(2, 2, i + 1)
    sns.boxplot(y=df[feature], color='lightcoral')
    plt.title(f"Boxplot of {feature}")
plt.tight_layout()
plt.show()

# 10. Pairplot of selected features
sns.pairplot(df[['Survived', 'Age', 'Fare', 'Sex', 'Pclass']], hue='Survived', palette='Set2')
plt.suptitle("Pairplot of Selected Features", y=1.02)
plt.show()

# 11. Correlation heatmap
plt.figure(figsize=(10, 6))
corr = df.corr()
sns.heatmap(corr, annot=True, cmap='coolwarm', fmt='.2f')
plt.title("Correlation Heatmap")
plt.show()

# 12. Interactive Plotly scatter plot (optional)
fig = px.scatter(
    df, x='Age', y='Fare', color='Survived',
    title='Age vs Fare colored by Survival',
    labels={'Survived': 'Survived'}
)
fig.show()

# 13. Inference examples
print("\nFeature-Level Inferences:")
print("- Females (Sex = 1) had a higher survival rate than males.")
print("- Passengers in Pclass 1 had higher survival chances.")
print("- Fare and Age show moderate correlation with survival.")
